core/src/main/java/com/valtech/aem/saas/core/bestbets/ClientBestBetsConsumerService.java

```diff
@@ -0,0 +1,165 @@
+package com.valtech.aem.saas.core.bestbets;
+
+import com.google.gson.Gson;
+import com.valtech.aem.saas.api.bestbets.BestBet;
+import com.valtech.aem.saas.api.bestbets.BestBetPayload;
+import com.valtech.aem.saas.api.bestbets.BestBetsActionFailedException;
+import com.valtech.aem.saas.api.bestbets.BestBetsConsumerService;
+import com.valtech.aem.saas.core.http.client.SearchRequestExecutorService;
+import com.valtech.aem.saas.core.http.request.SearchRequest;
+import com.valtech.aem.saas.core.http.request.SearchRequestDelete;
+import com.valtech.aem.saas.core.http.request.SearchRequestGet;
+import com.valtech.aem.saas.core.http.request.SearchRequestPost;
+import com.valtech.aem.saas.core.http.request.SearchRequestPut;
+import com.valtech.aem.saas.core.http.response.BestBetsDataExtractionStrategy;
+import com.valtech.aem.saas.core.http.response.ModifiedBestBetIdExtractionStrategy;
+import com.valtech.aem.saas.core.http.response.SearchResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import lombok.Builder;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.entity.ContentType;
+
```

Comment:
please add some javadoc

Reply:
In agreement with Nicolas, we agreed that javadoc is required only for classes that are exported (e.g. the interfaces and classes in api module)